### PR TITLE
Add message queue statistics

### PR DESCRIPTION
**Pull request by vlvkobal**
##### Summary
There can be a huge number of message queues, and they can be created and deleted quite frequently. So we need dynamic charts for them.
- [x] add message queue charts
- [x] make dimensions obsolete

Implements #1890

##### Component Name
`proc` plugin

### CODE REVIEW

**Comment by ilyam8** [comment]
@ktsaou please review, the PR adds dimension obsolete feature, i am interested  to have it merged

File: database/rrd.h [hunk 1]
```diff
@@ -242,21 +244,22 @@ struct rrddim {
...
+    RRDSET_FLAG_UPSTREAM_SEND       = 1 << 6, // if set, this chart should be sent upstream (streaming)
+    RRDSET_FLAG_UPSTREAM_IGNORE     = 1 << 7, // if set, this chart should not be sent upstream (streaming)
+    RRDSET_FLAG_UPSTREAM_EXPOSED    = 1 << 8, // if set, we have sent this chart definition to netdata master (streaming)
+    RRDSET_FLAG_STORE_FIRST         = 1 << 9, // if set, do not eliminate the first collection during interpolation
+    RRDSET_FLAG_HETEROGENEOUS       = 1 << 10, // if set, the chart is not homogeneous (dimensions in it have multiple algorithms, multipliers or dividers)
+    RRDSET_FLAG_HOMEGENEOUS_CHECK   = 1 << 11, // if set, the chart should be checked to determine if the dimensions as homogeneous
+    RRDSET_FLAG_HIDDEN              = 1 << 12, // if set, do not show this chart on the dashboard, but use it for backends
+    RRDSET_FLAG_SYNC_CLOCK          = 1 << 13, // if set, microseconds on next data collection will be ignored (the chart will be synced to now)
+    RRDSET_FLAG_OBSOLETE_DIMENSIONS = 1 << 14, // this is marked by the collector/module when a chart has obsolete dimensions
```
[the remark this answers is not in the document]
Comment (vlvkobal) [reply database/rrd.h]: It is already at the end. The block was just formatted.

File: database/rrd.h [hunk 2]
```diff
@@ -123,7 +124,8 @@ typedef struct rrdfamily RRDFAMILY;
 typedef enum rrddim_flags {
     RRDDIM_FLAG_NONE                            = 0,
     RRDDIM_FLAG_HIDDEN                          = (1 << 0),  // this dimension will not be offered to callers
-    RRDDIM_FLAG_DONT_DETECT_RESETS_OR_OVERFLOWS = (1 << 1)   // do not offer RESET or OVERFLOW info to callers
+    RRDDIM_FLAG_OBSOLETE                        = (1 << 1),  // this is marked by the collector/module as obsolete
+    RRDDIM_FLAG_DONT_DETECT_RESETS_OR_OVERFLOWS = (1 << 2)   // do not offer RESET or OVERFLOW info to callers
```
[the remark this answers is not in the document]
Comment (vlvkobal) [reply database/rrd.h]: Moved it.

File: database/rrd.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (vlvkobal) [reply database/rrd.h]: That's why I didn't get any errors while was testing the PR.

File: database/rrd.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (cakrit) [reply database/rrd.h]: I still see RRDDIM_FLAG_DONT_DETECT_RESETS_OR_OVERFLOWS after RRDDIM_FLAG_OBSOLETE
It was 1 << 1, now it is 1 << 2

**Comment by cakrit** [comment]
We did a detailed code review and some tests. It's fine.

**Comment by ktsaou** [comment]
ok, it seems good.

Please make another test: with this merged, run netdata under valgrind and somehow obsolete and delete a dimension. I want to make sure that the obsolete dimensions code, including the actual deletion, will not introduce any issues.

**Comment by vlvkobal** [comment]
I did tests with valgrind several times. The last one was done today after rebasing and pushing the last commit.

**Pull request merged**